checks/system/slurm/gres_gpu.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2016-2023 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+@rfm.simple_test
+class SlurmGPUGresTest(rfm.RunOnlyRegressionTest):
+    '''Ensure that the Slurm GRES (Gereric REsource Scheduling) of the number 
+       of gpus is correctly set on all the nodes of each partition.
+    '''
+
+    valid_systems = ['+scontrol +gpu']
```

Comment:
in theory, scontrol needs to be run only on the login nodes (no need for srun scontrol ...)

Reply:
I think the idea is to have it already like this for when we move to the firecrest scheduler.